__init__.py

```diff
@@ -64,11 +66,44 @@ def __init__(self):
         self.weather_api = OpenWeatherMapApi()
         self.weather_api.set_language_parameter(self.lang)
         self.platform = self.config_core["enclosure"].get("platform", "unknown")
+        self.gui_image_directory = Path(self.root_dir).joinpath("ui")
         self.weather_config = None
 
     def initialize(self):
         """Do these things after the skill is loaded."""
         self.weather_config = WeatherConfig(self.config_core, self.settings)
+        self.add_event("skill.weather.request-local", self.handle_get_local_weather)
+
+    def handle_get_local_weather(self, _):
+        """Handles a message bus command requesting current local weather information.
+
+        Such a request will typically come from a domain external to this skill that
+        requires weather information but should not go through the intent system
+        to get it.
+        """
+        system_unit = self.config_core.get("system_unit")
+        try:
+            weather = self.weather_api.get_weather_for_coordinates(
+                system_unit, self.weather_config.latitude, self.weather_config.longitude
+            )
+        except Exception:
+            self.log.exception("Unexpected error getting weather for skill API.")
+        else:
+            self._emit_local_weather_response(weather)
+
+    def _emit_local_weather_response(self, weather):
+        """Emits an event indicating that the request for local weather was satisfied.
+
+        Responds to the command for local weather retrieval.
+        """
+        image_path = self.gui_image_directory.joinpath(weather.current.condition.image)
+        weather_condition_url = "file://" + str(image_path)
+        event_data = dict(
+            temperature=weather.current.temperature,
+            weather_condition=weather_condition_url,
+        )
+        event = Message("skill.weather.local-retrieved", data=event_data)
```

Comment:
like the Alarm Skill - if we respond to the original message rather than emitting a new one, then we can use [`wait_for_response()`](https://github.com/MycroftAI/mycroft-messagebus-client/blob/master/mycroft_bus_client/client/client.py#L214) in the Homescreen Skill

Reply:
like the alarm skill, I am purposely trying to avoid using `wait_for_response()`.  this methodology is more representative
of the asynchronous event driven architecture.  it is my belief that this is the pattern we should be following whenever possible.

Reply:
related https://github.com/MycroftAI/mycroft-messagebus-client/pull/30